python/kwiver/vital/types/landmark.h

```diff
@@ -0,0 +1,196 @@
+/*ckwg +29
+ * Copyright 2017 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef PY_LANDMARK_H
+#define PY_LANDMARK_H
+
+#include <vital/types/landmark.h>
+
+#include <python/kwiver/vital/types/covariance_class.h>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/eigen.h>
+
+namespace py=pybind11;
+
+void landmark(py::module &m);
+
+class PyLandmarkBase
+{
+  public:
+
+    virtual ~PyLandmarkBase() = default;
+
+    virtual char get_type() { return '0';};
+
+    virtual py::object get_loc() { return py::none();};
+    virtual void set_loc(py::object) { };
+
+    virtual py::object get_scale() { return py::none();};
+    virtual void set_scale(py::object) { };
+
+    virtual py::object get_normal() { return py::none();};
+    virtual void set_normal(py::object) { };
+
+    virtual py::object get_covariance() { return py::none();};
+    virtual void set_covariance(py::object) { };
+
+    virtual kwiver::vital::rgb_color get_color() { return kwiver::vital::rgb_color();};
+    virtual void set_color(kwiver::vital::rgb_color) { };
+
+    virtual unsigned get_observations() { return 0;};
+    virtual void set_observations(unsigned) { };
+};
+
+class PyLandmarkd
+: public PyLandmarkBase
+{
+
```

Comment:
This file does not abide by the style manual with regard to indenting. Also private members usually follow the public members, even though this is not in the style manual.